nemo/collections/nlp/models/language_modeling/megatron_gpt_peft_models.py

```diff
@@ -368,6 +459,16 @@ def load_state_dict(self, state_dict, strict: bool = True):
         else:
             super().load_state_dict(state_dict, strict=True)
 
+    def on_load_checkpoint(self, checkpoint) -> None:
```

Comment:
why do we need to do this?

Reply:
there's some inconsistency in save/load logic after the distributed checkpoint PR. `on_load_checkpoint()` became the actual method to load checkpoints :( I talked to Eric about this. not sure what's the plan.

Reply:
We put the dist ckpt logic there for now but we'll probably refactor it a bit.